src/huggingface_hub/hf_file_system.py

```diff
@@ -122,7 +124,7 @@ def _repo_and_revision_exist(
     ) -> Tuple[bool, Optional[Exception]]:
         if (repo_type, repo_id, revision) not in self._repo_and_revision_exists_cache:
             try:
-                self._api.repo_info(repo_id, revision=revision, repo_type=repo_type)
+                self._api.repo_info(repo_id, revision=revision, repo_type=repo_type, timeout=HF_HUB_ETAG_TIMEOUT)
```

Comment:
This is not per-se a ETAG call (not a HEAD) but it is the proxy we use to confirm repo existence => serves the same purpose.